awx/api/serializers.py

```diff
@@ -4245,6 +4252,30 @@ def validate(self, attrs):
             for type_field_error in incorrect_type_fields:
                 error_list.append(_("Configuration field '{}' incorrect type, expected {}.").format(type_field_error[0],
                                                                                                     type_field_error[1]))
+        # Collect messages
+        messages = []
+        if attrs.get('messages', None) is not None and type(attrs['messages']) not in (dict, OrderedDict):
```

Comment:
I find this `if...elsif...` is a little verbose.  What about something like this?

```python
if 'messages' in attrs:
    if not isinstance(attrs, dict):
        error_list.append(...)
    else:
        for case in attrs['messages']:
            ...

Reply:
aside from needing to be `if not isinstance(attrs['messages'], dict):`, :100:

Reply:
totally agree - think I was trying to keep the McCabe complexity lower, but agree that the one-liner I came up with there is by itself too confusing